### ISSUE TITLE
Implement PCA

### ISSUE THREAD

**Issue by jithendaraa** [jithendaraa/visualise.ml#2]
Implement PCA which is the one of the most efficient dimensionality reduction methods:
Users should be able to:
- randomly initialize data of m samples and n features. (could implement a function in js just like [sklearn.datasets.make_regression](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_regression.html))
- Set n. if `n>3`, users have three options to reduce dimensions from: **n to 1**, **n to 2** or **n to 3**. No visualization needed for data when it is n-dimensional but should have visualisations on reducing n to 1,2 or 3.
- if `n<=3`, have an option to choose n'(final dimensions to which n is reduced to) and make the visualisation go from n-Dimensional(eg. 3d plot) to n'-Dimensional(eg. 2d plot). You could try changing the angle of vision such that the 3d visualisation changes to 2d or 3d similar to a camera tilt and causing a projection onto an arbitrary set of XY axes(for 2d) or X axis for 1d, such that the plane, through data projection captures the direction of maximum variance of the data.
- Cool animations
- Could give users the option to choose various parameters of the [PCA ](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html)function
- Get creative with your own features!
